From fb913c01d4af570050ec29f41503e0470d43d3a5 Mon Sep 17 00:00:00 2001
From: Linu Cherian <lcherian@marvell.com>
Date: Tue, 5 Feb 2019 20:44:10 +0530
Subject: [PATCH 0041/1921] octeontx2-af: Use the right wake_up API variant in
 CGX command interface

We use wait_event_timeout API for process waiting on command completion,
hence use wake_up API for waking up instead of wake_up_interruptible.

This improves the command completion waiting time.

Change-Id: I8a3ddebf8ed7e56183d6bcdbefadfb6e2e6a1a76
Signed-off-by: Linu Cherian <lcherian@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/cgx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
index 9b4cb17f186b..a2abbd5e9e63 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
@@ -595,7 +595,7 @@ static irqreturn_t cgx_fwi_event_handler(int irq, void *data)
 
 		/* Release thread waiting for completion  */
 		lmac->cmd_pend = false;
-		wake_up_interruptible(&lmac->wq_cmd_cmplt);
+		wake_up(&lmac->wq_cmd_cmplt);
 		break;
 	case CGX_EVT_ASYNC:
 		if (cgx_event_is_linkevent(event))
-- 
2.31.1

